src/main/java/hudson/tasks/test/AbstractTestResultAction.java

```diff
@@ -287,6 +316,84 @@ public TestResult findCorrespondingResult(String id) {
         return Collections.emptyList();
     }
 
+    /**
+     * Start for Test Result Trends.
+     * From here onwards start the code area responsible for generating various test result trends.
+     */
+
+    public hudson.tasks.junit.TestResult loadXml(){return null;}
+
+    /**
+     * A method for getting the list of packages for all levels of hierarchy.
+     * @return Array of packages of all hierarchies.
+     */
+    public String[] getProjectList(){
+
+        /*
+         * If project list is already ready no need to construct again.
+         */
+        if(projectList!=null) return projectList;
+        hudson.tasks.junit.TestResult r = loadXml();
```

Comment:
be more meaningful with var naming.  Variable name should be self explainable. `r` is too random here.